src/main/java/ch/hevs/cloudio/endpoint/CloudioPersistence.java

```diff
@@ -1,9 +1,19 @@
 package ch.hevs.cloudio.endpoint;
 
-import java.util.Set;
-
 public interface CloudioPersistence {
 
+    static class Message {
+        public long timestamp;
+        public String topic;
+        public byte[] data;
+
+        public Message(long timestamp, String topic, byte[] data) {
```

Comment:
Just a detail, but a second method `public Message(String topic, byte[] data)` that uses the current time as timestamp might be quite useful.

Reply:
True, did add it in 21e471c